src/stories/Header.stories.ts

```diff
@@ -1,26 +1,63 @@
 import type { Meta, StoryObj } from "@storybook/svelte";
-import Header from "$lib/components/Header.svelte";
+
+import Header from "$lib/components/Header";
+
+const ids: string[] = [];
+const genId = (): string => {
+  const maybeUniq = Math.random().toString(16).slice(2);
+  if (ids.includes(maybeUniq)) return genId();
+  else return maybeUniq;
+};
+
+const generateMenuItems = (n: number, prefix = "Menu 1") =>
+  [...Array(n)].map((_, i) => ({
+    id: genId(),
+    name: `${prefix} Child ${i + 1}`,
+    link: `/${prefix.replaceAll(" ", "-").toLowerCase()}-child-${i + 1}`,
+  }));
 
 const meta = {
   title: "Example/Header",
   component: Header,
-  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/7.0/react/writing-docs/docs-page
   tags: ["autodocs"],
-  parameters: {
-    // More on how to position stories at: https://storybook.js.org/docs/7.0/svelte/configure/story-layout
-    layout: "fullscreen",
-  },
 } satisfies Meta<Header>;
 
 export default meta;
 type Story = StoryObj<typeof meta>;
 
-export const LoggedIn: Story = {
+export const Default: Story = {
   args: {
-    user: {
-      name: "Jane Doe",
-    },
+    navItems: [
+      { id: genId(), name: "NavItem", link: "/nav-item" },
+      { id: genId(), name: "Active NavItem", link: "/active-nav-item", isCurrent: true },
+      {
+        id: genId(),
+        name: "Menu 1",
+        children: generateMenuItems(1),
+      },
+      {
+        id: genId(),
+        name: "Menu 2",
+        children: generateMenuItems(2, "Menu 2"),
+      },
+      {
+        id: genId(),
+        name: "MegaMenu 1",
+        megaMenuColumns: 2,
+        children: generateMenuItems(4, "MegaMenu 1"),
+      },
+      {
+        id: genId(),
+        name: "MegaMenu 2",
+        megaMenuColumns: 3,
+        children: generateMenuItems(7, "MegaMenu 2"),
+      },
+      {
+        id: genId(),
+        name: "MegaMenu 3",
+        megaMenuColumns: 5,
+        children: generateMenuItems(22, "MegaMenu 3"),
```

Comment:
In the storybook UI, it seems like when we're generating a mega menu, any modifications to `children` will not update the component preview. This is probably an issue with the `NavItem` implementation, and while I don't think it's an immediate problem, this could cause an issue during hydration if we're fetching content from the CMS on page load. 

I tried a few quick fixes by using reactive declarations, but unfortunately the results were either no change or caused mega menus to render as basic menus, and I haven't looked much further into why yet. 

I figure this is something we can maybe look at later down the line (if so I'll probably add a comment about it `NavItem`), but if y'all think it's worth a closer look now, I can definitely continue digging.

Also, somewhat related, if y'all have any ideas on how to make it easier to modify the nav items in the story, let me know. I'm not a huge fan of how it's just appearing as a bunch of JSON right now.

Reply:
Update: this was fixed by making `megaMenuColumns` reactive, which is something I had thought I had tried previously. Benaiah had done this in their review PR so doing something similar with 4abc75b got this working.